assets/src/pages/room/VideochatSection.tsx

```diff
@@ -119,6 +119,23 @@ const pinNewScreenShares = (
   screenSharingStreams.map((tile) => tile.mediaPlayerId).forEach(pinIfNotAlreadyPinned);
 };
 
+const manageOneOnOnePinning = (
+  mediaPlayerTiles: MediaPlayerTileConfig[],
+  { pinnedTileIds, pinIfNotAlreadyPinned, removePinnedEarlier }: PinningApi
+): void => {
+  if (mediaPlayerTiles.length !== 2) return;
+
+  const localUserTileId = mediaPlayerTiles.find((tile) => tile.streamSource === "local")?.mediaPlayerId;
+  const remoteUserTileId = mediaPlayerTiles.find((tile) => tile.streamSource === "remote")?.mediaPlayerId;
+  if (!localUserTileId || !remoteUserTileId) return;
+
+  const isLocalPinned = pinnedTileIds.includes(localUserTileId);
+  const isRemotePinned = pinnedTileIds.includes(remoteUserTileId);
+
+  if (!isLocalPinned && !isRemotePinned) pinIfNotAlreadyPinned(remoteUserTileId); // pinning new peer
+  if (isLocalPinned && isRemotePinned) removePinnedEarlier(localUserTileId, remoteUserTileId); // switching the peer
```

Comment:
I think it would be simpler to wrap pinning logic in function than using `removePinnedEarlier` mechanism. If there are only two tiles unpin the previous one and pin the current one.
```tsx
  const betterPinning = useCallback(
    (tileId: string) => {
      if (allTilesConfig.length === 2 && pinningApi.pinnedTileIds[0]) {
        pinningApi.unpin(pinningApi.pinnedTileIds[0]);
      }
      pinningApi.pin(tileId);
    },
    [allTilesConfig.length, pinningApi]
  );
  
<UnpinnedTilesSection
            tileConfigs={unpinnedTiles}
            showSimulcast={showSimulcast}
            oneColumn={isAnyTilePinned}
            webrtc={webrtc}
            pin={betterPinning}
            videoInVideo={pinnedTiles.length === 1}
            blockPinning={shouldBlockPinning}
            forceEncoding={forceEncoding}
          />
```
By doing so you could remove removePinnedEarlier and everything connected with that method.

Reply:
It solved some minor UI problems. This is great reviewing!